pandas/core/arrays/base.py

```diff
@@ -1352,7 +1352,11 @@ def _reduce(self, name: str, *, skipna: bool = True, **kwargs):
         ------
         TypeError : subclass does not define reductions
         """
-        raise TypeError(f"cannot perform {name} with type {self.dtype}")
+        meth = getattr(self, name, None)
+        if meth is None:
+            msg = f"'{type(self).__name__}' does not implement reduction '{name}'"
```

Comment:
Can you keep the self.dtype for the error message?

Reply:
(I would maybe also keep the "cannot perform" instead of "does not implement", as the latter sounds more as NotImplementedError, which in most cases here it's not the case about not being implemented, but about not being a valid operation (i.e. will never be implemented).

Reply:
Adding back in the dtype make sense.  The "does not implement" vs "cannot perform" I think the former is just more accurate.

Reply:
It might be "accurate" that we don't implement it, but its the message to the user that matters. I think it's more useful for them to know that the operation cannot be performed for a certain dtype, rather than that it is not implemented?

Reply:
compromise on "not supported"?